pkgs/development/tools/build-managers/bazel/default.nix

```diff
@@ -114,11 +114,10 @@ stdenv.mkDerivation rec {
         examples/java-native/src/test/java/com/example/myproject:hello
   '';
 
-  # Bazel expects gcc and java to be in the path.
   installPhase = ''
     mkdir -p $out/bin
     mv output/bazel $out/bin
-    wrapProgram "$out/bin/bazel" --prefix PATH : "${lib.makeBinPath [ stdenv.cc jdk ]}"
+    wrapProgram "$out/bin/bazel" --set JAVA_HOME "${jdk}"
```

Comment:
Why leave the wrapper for `JAVA_HOME`? By the same logic (which I agree with) `JAVA_HOME` should be set by the caller as well.

Reply:
Ah sorry I'm silly, this is for the `JAVA_HOME` used to run Bazel, not the one used to build java projects with Bazel.